react/components/ComponentsList/getComponents.ts

```diff
@@ -0,0 +1,101 @@
+import { has, path, pathOr } from 'ramda'
+import { ComponentsRegistry } from 'render'
+import { SidebarComponent } from './typings'
+
+const isSamePage = (page: string, pages: string[]) => (treePath: string) => {
+  if (treePath.startsWith(page)) {
+    return true
+  }
+
+  const currentPageLevel = page.split('/').length
+  const sameLevelPages = pages.filter(
+    (p: string) => p.split('/').length === currentPageLevel,
+  )
+
+  return !sameLevelPages.find((p: string) => treePath.startsWith(p))
+}
+
+const getParentContainerPropsGetter = (extensions: Extensions) => (
+  parentPath: string,
+) => {
+  return pathOr<string[], string[]>(
+    [],
+    [parentPath, 'props', 'elements'],
+    extensions,
+  )
+}
+
+const getComponentNameGetterFromExtensions = (extensions: Extensions) => (
+  treePath: string,
+) => extensions[treePath].component || ''
+
+const getComponentSchemaGetter = (
+  components: ComponentsRegistry | null,
+  extensions: Extensions,
+) => (treePath: string) => {
+  const getComponentName = getComponentNameGetterFromExtensions(extensions)
+  const component = getComponentName(treePath)
+  const getSchema = path([component, 'getSchema'], { ...components })
+  return (
+    path([component, 'schema'], { ...components }) ||
+    (typeof getSchema === 'function' && getSchema({}))
```

Comment:
Shouldn't `getSchema` be called with `extensions[treePath].props`? I thought that `{}` was just a fallback.